crypto/tls/tls.go

```diff
@@ -85,3 +114,31 @@ func (cfg *ClientConfig) GetGRPCDialOptions(enabled bool) ([]grpc.DialOption, er
 
 	return []grpc.DialOption{grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))}, nil
 }
+
+func mapCipherNameToID(cipherSuiteNames []string) ([]uint16, error) {
+	cipherSuites := []uint16{}
+	allCipherSuites := tlsCipherSuites()
+
+	for _, name := range cipherSuiteNames {
+		id, ok := allCipherSuites[name]
+		if !ok {
+			return nil, errors.New("unsupported cipher suite")
```

Comment:
I think it's more helpful if the error points out the problematic suite name:
```suggestion
			return nil, fmt.Errorf("unsupported cipher suite: %q", name)
```